### ISSUE TITLE
# type: ignore, Ignores entire dict instead of specific field/line

### ISSUE THREAD

**Issue by 0x1618** [microsoft/pyright#10710]
Adding `# type: ignore` causes the entire dictionary to be ignored instead of just the specified line or field, which can result in allowing an incorrect structure.

## Environment data

-   Pylance version: 2025.6.2
-   OS and version: MacOS 15.5
-   Python version (& distribution if applicable, e.g. Anaconda): 3.12.4

## Code Snippet

```python
from typing import TypedDict


class Schema(TypedDict):
    uid: int

    first_name: str

    history: dict[str, str]


def return_schema() -> Schema:
    return {
        "uid": 2,
        "history": {
            1: "abc"  # type: ignore (Cannot use numbers)
        },
        "first_name": "Maks",
        "it_should_be_flagged_but_it_is_not": True
    }

```

**Comment by erictraut** [comment]
Pyright is working correctly here — and in compliance with the Python typing spec, so this isn't a bug.

The `# type: ignore` comment suppresses any diagnostics whose range overlaps that line. In your example, there is one such diagnostic that would normally be reported on that line, so the `# type: ignore` suppresses it.

If you'd like to be more selective in which diagnostics pyright suppresses, you can use a `# pyright: ignore[xxx]` comment where `xxx` is a comma-delimited list of diagnostic checks that you'd like to filter.

For more details, refer to [this documentation](https://microsoft.github.io/pyright/#/comments?id=line-level-diagnostic-suppression) and [this section of the typing spec](https://typing.python.org/en/latest/spec/directives.html#type-ignore-comments).

**Comment by erictraut** [comment]
I'll also note that significant effort is put in to pyright to avoid "cascading errors" — the phenomenon where one error results in many other downstream errors. In this case, the error is contained to the specific place in the code that need to be fixed and does not result in other downstream errors. If you choose to ignore an error by suppressing it, then you will likely not see other errors.
